src/overlays/actors/ovl_En_Prz/z_en_prz.c

```diff
@@ -71,45 +93,442 @@ const ActorInit En_Prz_InitVars = {
+    this->actor.speedXZ = randPlusMinusPoint5Scaled(1.0f) + 4.0f;
+    func_80A75F18(this, 0);
+    this->unk_1EA = 1;
+    this->actionFunc = func_80A763E8;
+}
+
+void func_80A763E8(EnPrz* this, GlobalContext* globalCtx) {
+    Actor* sp3C = this->actor.parent;
+    Player* player = GET_PLAYER(globalCtx);
+    s16 yaw;
+    s32 pad;
+
+    if (&this->actor == this->unk_220) {
+        return;
+    }
+
+    if ((player->stateFlags1 & 0x8000000) && (this->unk_1F2 == 0)) {
+        func_80A76748(this);
+        return;
+    }
+
+    if (func_80A762C0(this, globalCtx)) {
+        func_80A76604(this, globalCtx);
+        return;
+    }
+
+    if ((this->unk_220 != NULL) && (this->unk_220->update == NULL)) {
+        this->unk_220 = this->actor.parent;
+    }
+
+    if ((this->unk_1EE == 0) && (this->unk_1F2 == 0)) {
+        s16 rot;
+
+        this->unk_214 = randPlusMinusPoint5Scaled(40.0f);
+
+        rot = BINANG_ROT180(this->unk_220->world.rot.y);
+        this->unk_1D8.x += Math_SinS(rot) * this->unk_214;
+        this->unk_1D8.y = randPlusMinusPoint5Scaled(40.0f) + (this->unk_220->world.pos.y + 40.0f);
+        rot = BINANG_ROT180(this->unk_220->world.rot.y);
+        this->unk_1D8.z += Math_CosS(rot) * this->unk_214;
+
+        this->unk_1EE = Rand_S16Offset(20, 30);
+    } else if (sqrtf(SQ(this->actor.world.pos.x - sp3C->world.pos.x) +
+                     SQ(this->actor.world.pos.z - sp3C->world.pos.z)) > 100.0f) {
+        Math_Vec3f_Copy(&this->unk_1D8, &sp3C->world.pos);
+    }
+
+    if (1) {}
+
+    if ((Rand_ZeroOne() < 0.1f) && (this->unk_1C8 == 0)) {
+        this->unk_1C8 = 1;
+    }
+
+    yaw = Math_Vec3f_Yaw(&this->actor.world.pos, &this->unk_1D8);
+    func_80A76070(this, yaw, globalCtx);
+}
+
+void func_80A76604(EnPrz* this, GlobalContext* globalCtx) {
+    this->unk_1EE = 0;
+    this->unk_1E4 = BINANG_ADD(this->actor.world.rot.y, 0x4000);
+    this->unk_1EA = 2;
+    this->actionFunc = func_80A76634;
+}
+
+void func_80A76634(EnPrz* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    func_80A76070(this, this->unk_1E4, globalCtx);
+
+    if (ABS_ALT(BINANG_SUB(this->actor.world.rot.y, this->unk_1E4)) < 0x100) {
+        if (func_80A762C0(this, globalCtx) != 0) {
+            this->unk_1E4 += 0x1500;
+            this->unk_1E4 += (s16)Rand_ZeroFloat(5000.0f);
+        } else if ((player->stateFlags1 & 0x8000000) && (player->actor.floorHeight < 30.0f)) {
+            this->actionFunc = func_80A763E8;
+        } else {
+            this->unk_1EE = 10;
+            this->unk_1EA = 1;
+            this->actionFunc = func_80A763E8;
+        }
+    }
+}
+
+void func_80A76748(EnPrz* this) {
+    this->actor.speedXZ = randPlusMinusPoint5Scaled(1.0f) + 3.0f;
+    this->unk_1EE = 0;
+    this->unk_1EA = 3;
+    this->skelAnime.playSpeed = 2.0f;
+    this->actionFunc = func_80A767A8;
+}
+
+typedef struct {
+    Actor actor;
+    char unk_144[0x184];
+    f32 unk_2C8;
+} EnPrzUnkParent;
+
```

Comment:
Could this just be using EnPr2? The structs seem relatively similar:
```
typedef struct EnPr2 {
    /* 0x0000 */ Actor actor;
    /* 0x0144 */ char unk_144[0x80];
    /* 0x01C4 */ EnPr2ActionFunc actionFunc;
    /* 0x01C8 */ char unk_1C8[0x100];
} EnPr2; // size = 0x2C8
```

Reply:
Looking at KZ, the parent actor to EnPrz seems to be `EnPr`.

Reply:
Thanks.